cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -362,6 +362,10 @@ private void restartIfNecessary(int podId, RestartContext restartContext)
         }
     }
 
+    private boolean isCrashlooping(Pod pod) {
+        return "CrashLoopBackOff".equals(pod.getStatus().getContainerStatuses().stream().filter(containerStatus -> containerStatus.getName().equals("kafka")).collect(Collectors.toList()).get(0).getState().getWaiting().getReason());
```

Comment:
This might suffer from NPEs (especially `.getState().getWaiting().getReason()`) and also possibly `IndexNotFoundException` if, for whatever reason the container couldn't be found. Probably better to code it more carefully.